sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -482,31 +424,79 @@ def _create_pipeline_workflow(self, args, pipeline, op_transformers=None, pipeli
         include_loop_task_refs.extend(with_loop_task)
       task_refs = include_loop_task_refs
 
-    # generate the Tekton Pipeline document
-    pipeline_template = {
+    # TODO: generate the PipelineRun template
+    pipeline_run = {
       'apiVersion': tekton_api_version,
-      'kind': 'Pipeline',
+      'kind': 'PipelineRun',
       'metadata': {
-        'name': pipeline.name or 'Pipeline',
-        'annotations': {
+        'name': sanitize_k8s_name(pipeline.name or 'Pipeline', suffix_space=4) + '-run',
+        'annotations': {  # TODO: do annotations apply to PipelineRun?
```

Comment:
Yes, annotations in pipelinerun do apply to all tasks which is the same behavior.